Mage.Sets/src/mage/cards/c/CourtOfArdenvale.java

```diff
@@ -0,0 +1,61 @@
+package mage.cards.c;
+
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.common.BeginningOfUpkeepTriggeredAbility;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.condition.common.MonarchIsSourceControllerCondition;
+import mage.abilities.decorator.ConditionalOneShotEffect;
+import mage.abilities.effects.common.BecomesMonarchSourceEffect;
+import mage.abilities.effects.common.ReturnFromGraveyardToBattlefieldTargetEffect;
+import mage.abilities.effects.common.ReturnFromGraveyardToHandTargetEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.ComparisonType;
+import mage.constants.TargetController;
+import mage.filter.FilterCard;
+import mage.filter.StaticFilters;
+import mage.filter.predicate.mageobject.ManaValuePredicate;
+import mage.target.common.TargetCardInYourGraveyard;
+
+/**
+ *
+ * @author Xanderhall
+ */
+public final class CourtOfArdenvale extends CardImpl {
+
+    private static final FilterCard filter = new FilterCard(StaticFilters.FILTER_CARD_PERMANENT);
```

Comment:
The copy constructor (`new FilterCard(FilterCard)`)should not be accessed outside the intended copy use I don't think. It is `public` currently but should be `protected` (and its argument `final`).

Anyway you want the filter message to be "permanent card with mana value 3 or less from your graveyard". You can use `new FilterPermanentCard(String)` for that.

Reply:
you are making a new `Filter` anyway (to add the mana value predicate), `new FilterPermanentCard()` has the exact same predicates as `StaticFilters.FILTER_CARD_PERMANENT.copy()` so there is no difference starting from there.

Reply:
And you want the full message to show on target selection "permanent card with mana value 3 or less from your graveyard". It will be displayed out of the filter there during gameplay.

Reply:
I have "permanent card" with the `ManaValuePredicate`, I assume that would build to the correct message when used with `TargetCardInYourGraveyard`

Reply:
![image](https://github.com/magefree/mage/assets/34709007/4000a429-a084-409a-87c7-1dc929eaa2be)

I'm talking about that choice box. It does use the filter's message directly.